Orange/widgets/visualize/owsieve.py

```diff
@@ -183,6 +176,8 @@ def set_data(self, data):
             self.attr_x = self.attr_y = None
             self.areas = []
             self.selection = set()
+        if self.data:
```

Comment:
I would move these lines a bit up, to the place where it originally was. Hence, you also can remove one `if`.